solana/pyth2wormhole/program/src/attest.rs

```diff
@@ -248,15 +262,30 @@ pub fn attest(ctx: &ExecutionContext, accs: &mut Attest, data: AttestData) -> So
     );
     solana_program::program::invoke(&transfer_ix, ctx.accounts)?;
 
+    let payload = batch_attestation.serialize().map_err(|e| {
+        trace!(&e.to_string());
+        ProgramError::InvalidAccountData
+    })?;
+
+    // Adjust message account size if necessary
+    if !accs.wh_message.is_initialized() && accs.wh_message.payload.len() != payload.len() {
```

Comment:
does realloc make sure that the account will remain rent exempt? 

I'm thinking that the size might be different if we have batches of different size and it can cause doing resizes more than necessary. 


minor: do we need to handle resizes? Can't we use a version field in the instruction input and `P2WMessageDrvData`? Or maybe a "{size}" as a seed?

Reply:
I don't think it takes care of rent exemptions. nice catch! I think wormhole would start complaining the next time we expand the format/symbol count per batch, which could become a nasty debug session. Thanks!

Reply:
Added a routine that makes sure exemption is maintained.